### PR TITLE
Fix broadcasting for empty sparse matrices

### PR DESCRIPTION
**Pull request by c42f**
For sparse empty broadcast result matrices, add special case to avoid dividing by zero when calculating the number of nonzeros.

Fixes issue reported on the mailing list: https://groups.google.com/forum/#!topic/julia-dev/s7jVxhEYC4M

Hold off merging for a few minutes and I'll write some tests as well.

### CODE REVIEW

**Comment by andreasnoack** [comment]
LGTM (when tests have been added)

**Comment by c42f** [comment]
Tests added.  I wasn't sure how exhaustive to be, but the generic broadcasting tests should cover most things so I ended up deleting most of what I wrote and just trying to cover both new code paths with a couple of cases.

**Comment by andreasnoack** [comment]
Thanks for fixing this.

File: base/sparse/sparsematrix.jl [hunk 1]
```diff
@@ -700,7 +700,8 @@ function gen_broadcast_body_sparse(f::Function, is_first_sparse::Bool)
         colptr1 = A_1.colptr; rowval1 = A_1.rowval; nzval1 = A_1.nzval
         colptr2 = A_2.colptr; rowval2 = A_2.rowval; nzval2 = A_2.nzval
 
-        nnzB = nnz(A_1) * div(B.n, A_1.n) * div(B.m, A_1.m)  +
+        nnzB = isempty(B) ? 0 :
```
Comment (tkelman) [inline base/sparse/sparsematrix.jl]: could we parenthesize this from `(isempty` to `A_1.m)) +` just to clarify where the ternary is covering?

Comment (c42f) [inline base/sparse/sparsematrix.jl]: The ternary has very low precedence so it's covering all the way up to `A_2.m`.  Will add parens in an attempt to make it clearer.

Comment (c42f) [inline base/sparse/sparsematrix.jl]: I pushed a tweak, hopefully it's clearer.

**Pull request merged**